File: intellij/src/main/kotlin/dev/nylo/plugin/project/NyloProjectDetector.kt
package dev.nylo.plugin.project

import com.intellij.openapi.project.Project
import java.io.File

object NyloProjectDetector {
    private val NYLO_FRAMEWORK_LINE = Regex("""^\s*nylo_framework\s*:""", RegexOption.MULTILINE)

    fun isNyloProject(project: Project): Boolean {
        val basePath = project.basePath ?: return false
        return isNyloProject(File(basePath))
    }

    fun isNyloProject(projectDir: File): Boolean {
        val pubspec = File(projectDir, "pubspec.yaml")
        if (!pubspec.isFile) return false
        return runCatching { pubspec.readText().contains(NYLO_FRAMEWORK_LINE) }.getOrDefault(false)
    }
}

File: vscode/src/core/project/nyloProjectDetector.ts
import * as fs from 'fs';
import * as path from 'path';

/**
 * Port of `dev.nylo.plugin.project.NyloProjectDetector`.
 *
 * A project is a Nylo project when its `pubspec.yaml` declares a `nylo_framework` dependency.
 * The regex requires the marker to start a line (after optional whitespace), so a `#`-commented
 * mention does not count.
 */
const NYLO_FRAMEWORK_LINE = /^\s*nylo_framework\s*:/m;

/** Pure predicate over the raw `pubspec.yaml` contents. */
export function isNyloPubspec(pubspecContents: string): boolean {
  return NYLO_FRAMEWORK_LINE.test(pubspecContents);
}

/** Whether the directory contains a Nylo `pubspec.yaml`. Returns false if missing/unreadable. */
export function isNyloProjectDir(projectDir: string): boolean {
  const pubspec = path.join(projectDir, 'pubspec.yaml');
  try {
    if (!fs.statSync(pubspec).isFile()) {
      return false;
    }
    return isNyloPubspec(fs.readFileSync(pubspec, 'utf8'));
  } catch {
    return false;
  }
}
